src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/ClassReader.java

```diff
@@ -2332,7 +2332,12 @@ public Void visitMethodSymbol(Symbol.MethodSymbol s, Void unused) {
                 thrown.add(addTypeAnnotations(thrownType, thrownType(i++)));
             }
             mt.thrown = thrown.toList();
-            mt.restype = addTypeAnnotations(mt.restype, TargetType.METHOD_RETURN);
+            /* possible information loss if the type of the method is void then we can't add type
```

Comment:
@vicente-romero-oracle Could we open a follow-up issue to add a standard way to retrieve these type annotations?
Type use annotations on constructors are valid Java >= 8 syntax and they are stored in the classfiles, so it would be nice to also get the annotations out of a classfile.

Reply:
That seems worth considering to me as a potential follow-up. One note is that as far as I can tell the current `javax.lang.model` implementation doesn't allow retrieving type annotations on constructors, I don't think this issue is specific to class loading.

Reply:
yes makes total sense

Reply:
@cushon Yes, at the moment we have to manually get to these annotations in the AST and from the classfile. Having a standard API for both source and bytecode would be nice.

Reply:
I have created: https://bugs.openjdk.org/browse/JDK-8320562